.github/workflows/tests.yml

```diff
@@ -64,85 +62,102 @@ jobs:
-            python-version: '3.8'
+          - python-version: '3.8'
+            default-channel: conda-forge
+          - test-type: unit
+            test-group: 3
     env:
-      OS: Windows
-      PYTHON: ${{ matrix.python-version }}
-      TEST_SPLITS: 3
-      TEST_GROUP: ${{ matrix.test-group }}
+      ALLURE_DIR: ${{ github.workspace }}\allure-results
+      PYTEST_MARKER: ${{ matrix.test-type == 'unit' && 'not integration' || 'integration' }}
+      PYTEST_NUMPROCESSES: '0'
+      PYTEST_SPLITS: ${{ matrix.test-type == 'unit' && '2' || '3' }}
+
     steps:
-      - uses: actions/checkout@v3
+      - name: Checkout Source
+        uses: actions/checkout@v4
         with:
           fetch-depth: 0
 
-      - name: Cache conda packages
+      - name: Hash + Timestamp
+        run: echo "HASH=${{ runner.os }}-${{ runner.arch }}-Py${{ matrix.python-version }}-${{ matrix.default-channel }}-${{ matrix.test-type }}-${{ matrix.test-group }}-$(date -u "+%Y%m")" >> $GITHUB_ENV
+        shell: bash
+
+      - name: Cache Conda
         uses: actions/cache@v3
-        env:
-          # Increase this value to reset cache
-          CACHE_NUMBER: 3
         with:
-          # setup.bat makes a directory junction from \conda_bin to
-          # C:\Miniconda; cache on top of the C:\Miniconda directory that exists
-          # at this stage.
-          path: C:\Miniconda\envs\conda-test-env\pkgs
-          key:
-            ${{ runner.os }}-conda-${{ env.CACHE_NUMBER }}-py${{ env.PYTHON }}-${{ matrix.default-channel }}-${{
-            hashFiles('recipe/meta.yaml', 'dev/windows/setup.bat', 'tests/requirements.txt') }}
-
-      - name: Set temp dirs correctly
-        # https://github.com/actions/virtual-environments/issues/712
-        run: |
-          echo "TMPDIR=$env:USERPROFILE\AppData\Local\Temp" >> $env:GITHUB_ENV
-          echo "TEMP=$env:USERPROFILE\AppData\Local\Temp" >> $env:GITHUB_ENV
-          echo "TMP=$env:USERPROFILE\AppData\Local\Temp" >> $env:GITHUB_ENV
-
-      - name: Setup environment for defaults
-        shell: cmd
-        if: matrix.default-channel == 'defaults'
-        run: |
-          call .\dev\windows\setup.bat
-
-      - name: Setup environment for conda-forge
-        shell: cmd
-        if: matrix.default-channel == 'conda-forge'
-        env:
-          CONDA_CHANNELS: ${{ matrix.default-channel }}
-          CONDA_CHANNEL_PRIORITY: strict
-        run: |
-          call .\dev\windows\setup.bat
+          path: ~/conda_pkgs_dir
+          key: cache-${{ env.HASH }}
 
-      - name: Python ${{ matrix.python-version }} on ${{ matrix.default-channel }}, ${{ matrix.test-type }} tests, group ${{ matrix.test-group }}
-        shell: cmd
+      - name: Setup Miniconda
+        uses: conda-incubator/setup-miniconda@v2
+        with:
+          condarc-file: .github\condarc-${{ matrix.default-channel }}
+          python-version: ${{ matrix.python-version }}
+          run-post: false  # skip post cleanup
```

Comment:
This'll run the tests in a non-base environment, are we sure that this won't be a problem with tests that assume to be running in base?

Reply:
looks like we're good, we seem to have naturally drifted away from this expectation